src/__tests__/Search/fetchRevisionByID.test.tsx

```diff
@@ -38,36 +34,37 @@ describe('Search View/fetchRevisionByID', () => {
       />,
     );
 
-    await screen.findAllByTestId('dropdown-select-base');
-    expect(screen.getAllByText('try')[0]).toBeInTheDocument();
-
     const searchInput = screen.getAllByRole('textbox')[0];
     await user.type(searchInput, 'abcdef123456');
-    await act(async () => void jest.runOnlyPendingTimers());
+    act(() => void jest.runAllTimers());
+    expect(global.fetch).toHaveBeenCalledWith(
+      'https://treeherder.mozilla.org/api/project/try/push/?revision=abcdef123456',
+      undefined,
+    );
 
     await user.clear(searchInput);
     await user.type(searchInput, 'abcdef1234567890abcdef1234567890abcdef12');
-    jest.runOnlyPendingTimers();
+    act(() => void jest.runAllTimers());
 
-    expect(global.fetch).toHaveBeenCalledWith(
-      'https://treeherder.mozilla.org/api/project/try/push/?revision=abcdef123456',
-    );
     expect(global.fetch).toHaveBeenCalledWith(
       'https://treeherder.mozilla.org/api/project/try/push/?revision=abcdef1234567890abcdef1234567890abcdef12',
+      undefined,
     );
-    await screen.findByText("you've got no arms left!");
-    await screen.findByText('What, ridden on a horse?');
+    expect(
+      await screen.findByText("you've got no arms left!"),
+    ).toBeInTheDocument();
+    expect(
+      await screen.findByText('What, ridden on a horse?'),
+    ).toBeInTheDocument();
   });
 
-  it('should reject fetchRevisionsByID if fetch returns no results', async () => {
-    global.fetch = jest.fn(() =>
-      Promise.resolve({
-        json: () => ({
-          results: [],
-        }),
+  it('should reject fetchRevisionByID if fetch returns no results', async () => {
+    (global.fetch as FetchMockSandbox).get(
+      'glob:https://treeherder.mozilla.org/api/project/*/push/*',
+      (url) => ({
+        results: url.includes('?revision') ? [] : getTestData().testData,
```

Comment:
We might be able to do it differently, but this works and is simple. Tell me if you'd like to explore other ways.